pkg/webservice/stream_limit.go

```diff
@@ -0,0 +1,128 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package webservice
+
+import (
+	"strconv"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/apache/yunikorn-core/pkg/common/configs"
+	"github.com/apache/yunikorn-core/pkg/log"
+)
+
+// StreamingLimiter tracks the concurrent streaming connections.
+type StreamingLimiter struct {
+	perHostStreams map[string]uint64 // number of connections per host
+	streams        uint64            // number of connections (total)
+
+	maxStreams        uint64 // maximum number of event streams
+	maxPerHostStreams uint64 // maximum number of event streams per host
+
+	sync.Mutex
+}
+
+func NewStreamingLimiter() *StreamingLimiter {
+	sl := &StreamingLimiter{
+		perHostStreams: make(map[string]uint64),
+	}
+
+	configs.AddConfigMapCallback("stream-limiter", func() {
```

Comment:
Will there only ever be a single streaming limiter? If so, this should be ok, but if there could be multiple, you need to generate a unique ID per limiter `stream-limiter-{seqnum}` or something would work. Also, you should probably have some sort of `Stop()` function that unregisters the callback.

Reply:
Yes, single instance. It's meant to live until Yunikorn's termination (it is created inside `init()`). I think it's fine, it's also not an issue for the MockScheduler. However, just to be safe, I do create a new callback id during each call and added a `Stop()` method for the unit tests.